google-cloud-vision/acceptance/google/cloud/vision/v1/image_annotator_feature_methods_test.rb

```diff
@@ -0,0 +1,92 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "minitest/autorun"
+require "minitest/spec"
+
+require "google/cloud/vision"
+
+describe "ImageAnnotatorFeatureMethods" do
+  it "detects faces using GCS" do
+    image_annotator_client = Google::Cloud::Vision.new
+    response = image_annotator_client.face_detection(
+      "gs://gapic-toolkit/President_Barack_Obama.jpg"
+    )
+    assert(response.face_annotations.size >= 1)
+  end
+
+  it "detects text from a local image" do
+    image_annotator_client = Google::Cloud::Vision.new
+    response = image_annotator_client.text_detection("acceptance/data/text.png")
+    assert(response.text_annotations.size >= 5)
+    assert(
+      response.full_text_annotation.text.include?(
+        "Google Cloud Client Library for Ruby"
+      )
+    )
+  end
+
+  it "detects a landmark from a local image" do
+    image_annotator_client = Google::Cloud::Vision.new
+    response = image_annotator_client.landmark_detection(
+      "acceptance/data/landmark.jpg"
+    )
+    assert(response.landmark_annotations.size >= 1)
+    assert(
+      response.landmark_annotations.map(&:description).include?(
+        "Mount Rushmore"
+      )
+    )
+  end
+
+  it "detects a face from a local image" do
+    image_annotator_client = Google::Cloud::Vision.new
+    response = image_annotator_client.face_detection("acceptance/data/face.jpg")
+    assert(response.face_annotations.size >= 1)
+  end
+
+  it "detects a logo from a local image" do
+    image_annotator_client = Google::Cloud::Vision.new
+    response = image_annotator_client.logo_detection("acceptance/data/logo.jpg")
+    assert(response.logo_annotations.size >= 1)
+    assert(
+      response.logo_annotations.map(&:description).include?(
+        "Google"
+      )
+    )
+  end
+
+  it "detects multiple features when none specified" do
+    image_annotator_client = Google::Cloud::Vision.new
+    response = image_annotator_client.annotate_image(
+      "gs://gapic-toolkit/President_Barack_Obama.jpg"
+    )
+    assert(response.face_annotations.size >= 1)
+    assert(response.label_annotations.size >= 1)
+    assert(response.crop_hints_annotation.crop_hints.size >= 1)
+  end
+
+  it "detects multiple specified features" do
+    image_annotator_client = Google::Cloud::Vision.new
+    response = image_annotator_client.annotate_image(
+      "gs://gapic-toolkit/President_Barack_Obama.jpg",
+      features: [{ type: :FACE_DETECTION }, { type: :CROP_HINTS }]
```

Comment:
Annotating an image with multiple *specific features* is a common task. Would it be possible to simplify this syntax a bit, eg. flatten the `features:` named parameter down to named parameters per feature?

eg.

```ruby
vision.annotate_image "foo.png", face_detection: true, crop_hints: true
```

or flatten the features objects and just read their TYPE?

```ruby
vision.annotate_image "foo.png", features: [:face_detection, :crop_hints]
```

/cc @frankyn for Dev X thoughts

Reply:
The helpers make it easy to annotate an image with a **single feature** or **all features** but the syntax for _**multiple features**_ is ... kinda gross 😋

Reply:
I can certainly add the syntactic sugar to make this look like either of the samples you suggest, but I worry that could leave us in a sticky place if the Vision API adds fields to the `google.cloud.vision.v1.Feature` message, since there would be no way to associate them with the feature type.

@lukesneeringer Do you know why the Vision Feature proto message is structured this way? Is there a potential for fields to be added that will commonly be specified? If not, I will go ahead and make the change that @remi suggests.

Reply:
/cc @surekhamoturu

Reply:
I am not sure how much `max_results` really matters, [but `model` has the potential to be common and significant](https://github.com/googleapis/googleapis/blob/master/google/cloud/vision/v1/image_annotator.proto#L99). It is pretty easy to conceive of a situation where someone wants to get multiple features and use a distinct model per feature (which is why it was ...erm...modeled (wah wah) that way).

Would it be possible to use the second of @remi's constructs and _either_ take a symbol _or_ a Message/Hash? In other words, nothing would stop you from doing...

```ruby
vision.annotate_image "foo.png", features: [:face_detection, Feature { type: CROP_HINTS, model: "builtin/latest" }]
```

Reply:
> Would it be possible to use the second of @remi's constructs and either take a symbol or a Message/Hash?

Yep, making the symbol "shorthand" for constructing a full feature seems like an OK compromise position, as long as that's not confusing to users. (I don't think it will be.)

Reply:
Re-formatting Luke's suggestion into Ruby syntax:

```ruby
vision.annotate_image "foo.png", features: [ :face_detection, { type: :crop_hints, max_results: 20 }]
```

Reply:
Ruby idiom note: please allow for lowercase symbols 😄 

I like Luke's suggestion! `feature:` is array of symbols & Hashes

Reply:
👍

Reply:
Done, following @lukesneeringer's suggestion; thanks, Luke!

Reply:
👍  inclusion in guide, can do